### PR TITLE
Cover: Move overlay and opacity controls to color panel

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Related:

- https://github.com/WordPress/gutenberg/pull/40084

Depends on:

- https://github.com/WordPress/gutenberg/pull/41091

## What?

Moves the Cover block's overlay color and opacity controls into the inspector controls sidebar's Color panel.

## Why?

Makes the Cover block's color-related controls appear under the same panel most blocks' do for consistency.

## How?

Renders the Cover block's color-related controls via the grouped inspector controls slot so they are rendered alongside any color block supports and under the Color panel in the sidebar.

## Testing Instructions

1. Edit a post, add a cover block, and select it
2. Make sure that the Cover block's overlay color and opacity controls behave as expected
3. Test that the overlay and opacity controls can be reset via the Color panel's menu

## Screenshots or screencast

Screen Shot 2022-05-17 at 5 37 56 pm

### CODE REVIEW

**Comment by aaronrobertshaw** [comment]
I've rebased this onto https://github.com/WordPress/gutenberg/pull/41091 and [tweaked accordingly](https://github.com/WordPress/gutenberg/pull/41102/commits/c51823139604783a33d8460ab6b4427ae9ce169a). It should be right to review as a follow-up once #41091 lands.

**Comment by noisysocks** [comment]
With the design in `trunk` it's quite clear that the _Opacity_ slider will adjust the overlay's opacity because the slider appears in the _Overlay_ panel. With this change, though, it's less clear to me that the _Opacity_ slider has to do with the overlay. 

Would be good to get a second opinion from a designer on this but should we move the the opacity slider to within the colour picker which is where it is in other blocks?

https://user-images.githubusercontent.com/612155/169430721-3afbbb56-73d2-4016-b9ed-85992fee32df.mp4

**Comment by aaronrobertshaw** [comment]
Thanks for the review @noisysocks 👍 

> With the design in trunk it's quite clear that the Opacity slider will adjust the overlay's opacity because the slider appears in the Overlay panel. With this change, though, it's less clear to me that the Opacity slider has to do with the overlay.

Agreed. The reasoning for injecting this into the Color supports panel was that there are also plans to add text color support to the Cover block which would result in two color related panels in the sidebar. Also, on trunk the panel's reset all menu item doesn't reset the cover block.

> Would be good to get a second opinion from a designer on this but should we move the the opacity slider to within the colour picker which is where it is in other blocks?

This is on my list to explore.  The color picker allows transparency via the `enableAlpha` prop though I'm not sure how this will play with the desire to have a default opacity etc. Without looking into it yet, I'm also unsure as to how we'd migrate existing blocks and maintain the CSS classes e.g. `has-background-dim-<value>`.

**Comment by noisysocks** [comment]
> The reasoning for injecting this into the Color supports panel was that there are also plans to add text color support to the Cover block which would result in two color related panels in the sidebar.

That makes sense. It think it makes it even more important that the slider moves into the colour picker, then, because otherwise it won't be clear if the slider adjusts the text opacity or overlay opacity.

**Review by jorgefilipecosta — COMMENTED** [review]
Hi @noisysocks what if we name "Opacity" as "Overlay Opacity" to make it clear it refers to the overlay?
There are some reasons to have opacity as a separate control instead of as the transparency value of the color:
- We can easily set a default opacity as we do now.
- If the is a complex preset gradient with 6 color stops, and we want to make that gradient have 20% opacity it is easier to just select the opacity in a single place, than changing the 6 color stops to achieve the 20% opacity.
- If I want to use a preset color or gradient but just change the opacity, using separate opacities we only rely on classes we still use the preset classes if we go for the color transparency approach we would use an inline color.

**Comment by aaronrobertshaw** [comment]
> There are some reasons to have opacity as a separate control instead of as the transparency value of the color

Those are some great points. 

I think renaming the control is a nice, simple compromise. It also keeps this progressing forward to where we can explore opting into text color block support for the Cover block without having two color related panels.

I've renamed the control from "Opacity" to "Overlay opacity" in https://github.com/WordPress/gutenberg/pull/41102/commits/6b05ca5bec08f00c380c4dc240c2b27e17e76722.

**Comment by ramonjd** [comment]
Working great for me! 

There's one thing I noticed: I think we have to set the `dimRatio` to the default `50` when resetting the color properties. I'm seeing a classname of `has-background-dim-NaN` on the `.wp-block-cover__background` element.

To replicate:

1. Create a new cover block
2. Reset the color properties
3. Duplicate the block.
4. See that the duplicated cover block's background is black with opacity:1

https://user-images.githubusercontent.com/6458278/172076225-80c192cd-60c5-46bf-b412-ab2df9fe95b7.mp4

I checked on trunk and I couldn't reproduce.

**Comment by aaronrobertshaw** [comment]
Thanks for catching that @ramonjd 👍 

I've tweaked the appropriate callbacks for the opacity control (https://github.com/WordPress/gutenberg/pull/41102/commits/f4009075f7178115e01834f5ae2ab8852653713c). I believe the opacity is now correct when you duplicate a Cover block that has background media.

**Review by ramonjd — APPROVED** [review]
Retested and LGTM!

**Comment by glendaviesnz** [comment]
Tests well for me, but I wonder if it is confusing to have the separate opacity option, and the alpha channel in the custom color picker:
custom-color

It is [explicitly toggled on here](https://github.com/WordPress/gutenberg/pull/41102/files#diff-4af4f90d0daaf9ff4f7c1836fe266febc31fac60488d8084b942cb7f62cb2a7bR249), so maybe there is a reason for having both that I am overlooking?

**Comment by aaronrobertshaw** [comment]
Thanks for raising that issue @glendaviesnz 👍 

I've removed the alpha channel from the overlay color picker (https://github.com/WordPress/gutenberg/pull/41102/commits/4667be190ecc95747579ae3e70f4843f19340b5c). It was an oversight of mine while copy and pasting in the color dropdown code.

Screen Shot 2022-06-07 at 10 59 30 am

**Review by glendaviesnz — APPROVED** [review]
LGTM

**Pull request merged**
